Automattic/jetpack
# Subscriptions: Add unit tests and fixtures to validate block content parsing

### Changes proposed in this Pull Request:
This PR adds unit and block fixture parsing tests to the Subscriptions block, *and* extricate the controls into a `controls.js` file. 

No block functionality is affected.

### Jetpack product discussion
p1HpG7-b3G-p2

###  Does this pull request change what data or activity we track or use?
No.

### Testing instructions:

Running the unit tests:

`cd projects/plugins/jetpack && yarn fixtures:test extensions/blocks/subscriptions/test/`

We should also ensure that the block works as expected in the editor and saved stats on a site.

To test the subscriptions block in a browser we need an active connection or have the plugin running on WPCOM in order to activate the module. 

Apply D57958-code to your sandbox and add a subscriptions block to a new sandboxed test site.

<img width="832" alt="Screen Shot 2021-03-04 at 11 53 21 am" src="https://user-images.githubusercontent.com/6458278/109895392-401d1400-7ce3-11eb-85d7-92b0cb432931.png">
<img width="735" alt="Screen Shot 2021-03-04 at 12 15 40 pm" src="https://user-images.githubusercontent.com/6458278/109895483-6347c380-7ce3-11eb-9e24-6eb88d0cb89d.png">


### Proposed changelog entry for your changes:
None. No functional changes made.
